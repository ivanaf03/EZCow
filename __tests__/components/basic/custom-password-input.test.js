import React from "react";

import {render, fireEvent} from "@testing-library/react-native";

import CustomPasswordInput from "../../../components/basic/custom-password-input";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null
}));

describe("CustomPasswordInput", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(<CustomPasswordInput text="Test" placeholder="Test" value="Test" />);
        expect(tree).toMatchSnapshot();
    });

    it("should change value when typing", () => {
        const onChangeText = jest.fn();
        const tree = render(<CustomPasswordInput text="Test" placeholder="Test" value="Test" onChangeText={onChangeText} />);
        fireEvent.changeText(tree.getByPlaceholderText("Test"), 'Test2');
        expect(onChangeText).toHaveBeenCalledTimes(1);
    });

    it("should show password when clicking on eye icon", () => {
        const onChangeText = jest.fn();
        const tree = render(<CustomPasswordInput text="Test" placeholder="Test" value="Test" onChangeText={onChangeText} />);
        const eyeIcon = tree.getByTestId("eye-icon");
        expect(tree.getByPlaceholderText("Test").props.secureTextEntry).toBe(true);
        fireEvent.press(eyeIcon);
        expect(tree.getByPlaceholderText("Test").props.secureTextEntry).toBe(false);
    });

    it("should hide password when clicking on eye icon", () => {
        const onChangeText = jest.fn();
        const tree = render(<CustomPasswordInput text="Test" placeholder="Test" value="Test" onChangeText={onChangeText} />);
        const eyeIcon = tree.getByTestId("eye-icon");
        expect(tree.getByPlaceholderText("Test").props.secureTextEntry).toBe(true);
        fireEvent.press(eyeIcon);
        expect(tree.getByPlaceholderText("Test").props.secureTextEntry).toBe(false);
        fireEvent.press(eyeIcon);
        expect(tree.getByPlaceholderText("Test").props.secureTextEntry).toBe(true);
    });

});