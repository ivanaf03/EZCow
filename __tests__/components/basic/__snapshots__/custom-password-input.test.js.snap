// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPasswordInput should render correctly 1`] = `
<View
  style={
    [
      {
        "marginTop": -4,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Light",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    Test
    :
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    <TextInput
      placeholder="Test"
      secureTextEntry={true}
      style={
        [
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          {
            "marginRight": 8,
          },
          {
            "backgroundColor": "#ffffff",
          },
          {
            "borderBottomWidth": 2,
            "borderLeftWidth": 2,
            "borderRightWidth": 2,
            "borderTopWidth": 2,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
          },
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
      value="Test"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="eye-icon"
    />
  </View>
</View>
`;
